src/Microsoft.AspNetCore.Sockets.Client/ITransportFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public interface ITransportFactory
+    {
+        ITransport CreateTransport(TransportType availableServerTransports, HttpClient httpClient);
```

Comment:
Because it would require always providing the HttpClient to Start method or we would use different http clients. Right now you can just provide the factory and if you don't provide http client we will give you the one we created so that the same client is being used across the stack.